File: src/main/java/name/seguri/java/leetcode/_704/Solution.java
package name.seguri.java.leetcode._704;

class Solution {

  /**
   *
   *
   * <ul>
   *   <li>nums.length > 0
   * </ul>
   */
  public int search(int[] nums, int target) {
    var lo = 0;
    var hi = nums.length - 1;
    while (lo <= hi) {
      var mid = lo + (hi - lo) / 2;
      if (target < nums[mid]) {
        hi = mid - 1;
      } else if (target > nums[mid]) {
        lo = mid + 1;
      } else {
        return mid;
      }
    }
    return -1;
  }
}

File: src/main/kotlin/name/seguri/kotlin/leetcode/_704/Solution.kt
package name.seguri.kotlin.leetcode._704

class Solution {
  fun search(nums: IntArray, target: Int): Int {
    var lo = 0
    var hi = nums.size - 1
    while (lo <= hi) {
      val mid = lo + (hi - lo) / 2
      when {
        target > nums[mid] -> lo = mid + 1
        target < nums[mid] -> hi = mid - 1
        else -> return mid
      }
    }
    return -1
  }
}
